### PR TITLE
bug(Google AI Gemini) — fix mapping for tools with parameters with nested object structures

### PR DESCRIPTION
**Pull request by glaforge**
* bug(Google AI Gemini) — fix mapping for tools with parameters with nested object structures
* fix(JsonSchemas) — minor improvement to cover all cases of strings

### CODE REVIEW

File: langchain4j-google-ai-gemini/src/test/java/dev/langchain4j/model/googleai/GoogleAiGeminiChatModelIT.java [hunk 1]
```diff
@@ -43,7 +43,7 @@
 
 public class GoogleAiGeminiChatModelIT {
 
-    private static final String GOOGLE_AI_GEMINI_API_KEY = System.getenv("GOOGLE_AI_GEMINI_API_KEY");
+    private static final String GOOGLE_AI_GEMINI_API_KEY = System.getenv("GEMINI_AI_KEY");
```
Comment (langchain4j) [inline langchain4j-google-ai-gemini/src/test/java/dev/langchain4j/model/googleai/GoogleAiGeminiChatModelIT.java]: Please keep it `GOOGLE_AI_GEMINI_API_KEY` or let's change it in CI builds as well

**Review by ghost — DISMISSED** [review]
@glaforge thank you!

File: langchain4j-google-ai-gemini/src/test/java/dev/langchain4j/model/googleai/GoogleAiGeminiChatModelIT.java (hunk 1 above)
Comment (glaforge) [reply langchain4j-google-ai-gemini/src/test/java/dev/langchain4j/model/googleai/GoogleAiGeminiChatModelIT.java]: Sorry, fixed!

**Review by ghost — APPROVED** [review]
@glaforge thank you, great job!

**Pull request merged**